### PR TITLE
Nfdiv 3745 missing notification

### PR DESCRIPTION
**Pull request by nigeldunne**
### Change description ###

Missing notification: email to notify applicant 2 that applicant 1 has submitted an intent to switch to sole is not being sent to applicant 2's solicitor

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/NFDIV-3745

### Pull request checklist ###

**Before raising**
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] README and other documentation has been updated / added (if needed)

**Before merging**
- [ ] this ticket been reviewed by QA
- [ ] the user story been signed off by the PO

**Note:** Bug fixes, dependency updates and technical tasks do not directly impact the user experience and can be merged without QA and PO review.

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/citizen/notification/Applicant1IntendToSwitchToSoleFoNotification.java [hunk 1]
```diff
@@ -40,4 +46,18 @@ public void sendToApplicant2(CaseData data, Long id) {
             data.getApplicant2().getLanguagePreference()
         );
     }
+
+    @Override
+    public void sendToApplicant2Solicitor(CaseData data, Long id) {
+        if (YES.equals(data.getFinalOrder().getDoesApplicant1IntendToSwitchToSole())) {
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/citizen/notification/Applicant1IntendToSwitchToSoleFoNotification.java:52]: do you even get to this point if they say no?

File: src/main/java/uk/gov/hmcts/divorce/notification/SwitchToSoleSolicitorTemplateContent.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    CommonContent commonContent;
+
+    public static final String APPLICANT_1_NAME = "applicant 1 name";
+    public static final String APPLICANT_2_NAME = "applicant 2 name";
+
+    public Map<String, String> templatevars(CaseData caseData, Long caseId, Applicant applicant, Applicant partner) {
+        Map<String, String> templateContent = commonContent.mainTemplateVars(caseData, caseId, applicant, partner);
+
+        templateContent.put(APPLICANT_1_NAME,
+            join(" ", caseData.getApplicant1().getFirstName(), caseData.getApplicant1().getLastName()));
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/notification/SwitchToSoleSolicitorTemplateContent.java:36]: use applicant.getFullName()

Comment (nigeldunne) [reply src/main/java/uk/gov/hmcts/divorce/notification/SwitchToSoleSolicitorTemplateContent.java]: Done.

File: src/main/java/uk/gov/hmcts/divorce/citizen/notification/Applicant1IntendToSwitchToSoleFoNotification.java (hunk 1 above)
Comment (nigeldunne) [reply src/main/java/uk/gov/hmcts/divorce/citizen/notification/Applicant1IntendToSwitchToSoleFoNotification.java]: It doesn't look like it, removed

Comment (nigeldunne) [reply src/main/java/uk/gov/hmcts/divorce/citizen/notification/Applicant1IntendToSwitchToSoleFoNotification.java]: It looke like it's only called from CitizenIntendToSwitchToSoleFO and the decision is made there

**Review by linusnorton — APPROVED** [review]

**Pull request merged**